File: Solutions/2300 - 2399/2399. Check Distances Between Same Letters/Solution.java
class Solution 
{
    public boolean checkDistances(String s, int[] distance) 
    {
        int[] indices = new int[26];
        Arrays.fill(indices, -1);

        for(int i=0; i<s.length(); i++)
        {
            char curr = s.charAt(i);
            int indx = curr - 'a';

            if(indices[indx] == -1)
            {
                indices[indx] = i;
            }
            else
            {
                int dist1 = i - indices[indx] - 1;
                if(dist1 != distance[indx])
                {
                    return false;
                }                
            }
        }
        return true;
    }
}

File: Solutions/2300 - 2399/2399. Check Distances Between Same Letters/Solution.py
class Solution:
    def checkDistances(self, s: str, distance: List[int]) -> bool:
        indices = {}

        for i, ch in enumerate(s):
            if ch not in indices:
                indices[ch] = i
            else:
                dist1 = i - indices[ch] - 1
                dist2 = distance[ord(ch) - ord('a')]
                if dist1 != dist2:
                    return False
        
        return True
